addons/website_sale/controllers/main.py

```diff
@@ -1770,6 +1770,13 @@ def shop_payment_validate(self, sale_order_id=None, **post):
 
         if order and not order.amount_total and not tx_sudo:
             order.with_context(send_email=True).with_user(SUPERUSER_ID).action_confirm()
+            auto_invoice = order.env['ir.config_parameter'].get_param('sale.automatic_invoice')
+            # create an invoice for order with zero total amount and automatic invoice enabled
+            if auto_invoice and order.reward_amount:
```

Comment:
`reward_amount` is defined in `sale_loyalty`, you cannot use that field in `website_sale`.